test/smoke/dashCore.js

```diff
@@ -1,18 +1,106 @@
+  // Verify it on masternodes, wallet-nodes, full-nodes, miners
+  it('should have correct blockhash and blocks count', async () => {
+    const blockHashes = {};
+    for (let i = 0; i < nodeNames.length; i++) {
+      const config = {
+        protocol: 'http',
+        user: networkConfig.variables.dashd_rpc_user,
+        pass: networkConfig.variables.dashd_rpc_password,
+        host: networkConfig.inventory._meta.hostvars[nodeNames[i]].public_ip,
+        port: networkConfig.variables.dashd_rpc_port,
+      };
+      const rpc = new RpcClient(config);
+      const { result: { blocks, bestblockhash } } = await rpc.getBlockchainInfo();
+      if (!blockHashes[blocks]) {
+        blockHashes[blocks] = bestblockhash;
+      }
+      expect(blockHashes[blocks]).to.be.equal(bestblockhash);
+    }
+    const blocksCounts = Object.keys(blockHashes);
+    expect(Math.max(...blocksCounts)
+      - Math.min(...blocksCounts)).to.be.below(3);
   });
+});
+
+describe('Masternodes', () => {
+  networkConfig.inventory.masternodes.hosts.forEach((nodeName) => {
+    describe(nodeName, () => {
+      const config = {
+        protocol: 'http',
+        user: networkConfig.variables.dashd_rpc_user,
+        pass: networkConfig.variables.dashd_rpc_password,
+        host: networkConfig.inventory._meta.hostvars[nodeName].public_ip,
+        port: networkConfig.variables.dashd_rpc_port,
+      };
+      const rpc = new RpcClient(config);
+      // masternode status
+      it('should masternodes be enabled', async () => {
+        const masternodelist = await rpc.masternodelist();
+        const idNodes = Object.keys(masternodelist.result);
+        const masterIps = [];
+        for (let i = 0; i < idNodes.length; i++) {
+          masterIps.push(masternodelist.result[idNodes[i]].address.split(':')[0]);
+          expect(masternodelist.result[idNodes[i]].status).to.be.equal('ENABLED');
+        }
+        const masterIpsInv = [];
+        for (const nName of networkConfig.inventory.masternodes.hosts) {
+          masterIpsInv.push(networkConfig.inventory._meta.hostvars[nName].public_ip);
+        }
+        expect(masterIps.sort()).to.deep.equal(masterIpsInv.sort());
+      });
+    });
+  });
+
   describe('Miners', () => {
     // waitfornewblock
-    it('should mine blocks');
+    it('should mine blocks', async function () {
+      this.timeout(160000);
+      const config = {
+        protocol: 'http',
+        user: networkConfig.variables.dashd_rpc_user,
+        pass: networkConfig.variables.dashd_rpc_password,
+        host: networkConfig.inventory._meta.hostvars[
+          networkConfig.inventory.masternodes.hosts[0]].public_ip,
+        port: networkConfig.variables.dashd_rpc_port,
+      };
+      const rpc = new RpcClient(config);
+      const { result } = await rpc.getBlockCount();
+      const { result: { height } } = await rpc.waitForNewBlock(150000);
+      expect(result + 1).to.be.equal(height);
```

Comment:
This check is unnecessary. `waitForNewBlock` timeout should be more than test timeout.

Reply:
waitForNewBlock always returns _current block if new block was not mined during wait time. we should leave this verification

Reply:
There are two valid approaches:
1. Wait for a new block with timeout bigger than test timeout.
2. Wait for new block with timeout less than test timeout and then check block height.

I talked about the first one. You implemented the second one. Let's keep your version but please add change assertion to make test more descriptive:
```
expect(height).to.be.above(previousHeight, 'no new blocks');
```

Reply:
if wait for a new block with timeout bigger than test timeout and no new block test stuck. test timeout should be bigger then waitForNewBlock timeout

Reply:
Ahhh... because of `await`, right.